firehose/firehose_kinesis.go

```diff
@@ -3,24 +3,28 @@ package firehose
 import (
 	"encoding/json"
 	"fmt"
+	"os"
 	"sync/atomic"
 	"time"
 
-	"github.com/Clever/heka-clever-plugins/aws"
 	"github.com/Clever/heka-clever-plugins/batcher"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	awsFirehose "github.com/aws/aws-sdk-go/service/firehose"
 )
 
+// FirehoseWriter writes record batches to a firehose stream
 type FirehoseWriter struct {
-	conf     *FirehoseWriterConfig
-	batchers map[string]batcher.Batcher
+	streamName     string
+	messageBatcher batcher.Batcher
+	firehoseClient *awsFirehose.Firehose
 
-	mockEndpoint string
-
-	recvRecordCount    int64
-	sentRecordCount    int64
-	droppedRecordCount int64
+	recvRecordCount   int64
+	sentRecordCount   int64
+	failedRecordCount int64
 }
 
+// FirehoseWriterConfig is the set of config options used in NewFirehoseWriter
 type FirehoseWriterConfig struct {
 	// The value of this field is used as the firehose stream name
 	StreamName string
```

Comment:
nit: these field name comments don't follow the golint convention of `// <FieldName> ...`